pkg/controller/events/node_execution.go

```diff
@@ -0,0 +1,84 @@
+package events
+
+import (
+	"context"
+	"strings"
+
+	"github.com/flyteorg/flyteidl/clients/go/events"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/event"
+	"github.com/flyteorg/flytepropeller/pkg/controller/config"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/flyteorg/flytestdlib/storage"
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+//go:generate mockery -all -output=mocks -case=underscore
+
+// Recorder for Node events
+type NodeEventRecorder interface {
+	RecordNodeEvent(ctx context.Context, event *event.NodeExecutionEvent, eventConfig *config.EventConfig) error
+}
+
+type nodeEventRecorder struct {
+	eventRecorder events.NodeEventRecorder
+	store         *storage.DataStore
+}
+
+// In certain cases, a successful node execution event can be configured to include raw output data inline. However,
+// for large outputs these events may exceed the event recipient's message size limit, so we fallback to passing
+// the offloaded output URI instead.
+func (r *nodeEventRecorder) handleFailure(ctx context.Context, ev *event.NodeExecutionEvent, err error, eventConfig *config.EventConfig) error {
+	// Only attempt to retry sending an event in the case we tried to send raw output data inline
+	if eventConfig.RawOutputPolicy != config.RawOutputPolicyInline || len(ev.GetOutputUri()) > 0 {
+		return err
+	}
+	st, ok := status.FromError(err)
+	if !ok || st.Code() != codes.ResourceExhausted {
+		// Error was not a status error
+		return err
+	}
+	if !strings.HasPrefix(st.Message(), "message too large") {
```

Comment:
There is no error code for this? :(

Reply:
the code is "ResourceExhausted" but i think that encompasses several causes :(

Reply:
lets add a comment that in the case the code is resource exhausted, it could be because of the size. Also do we know the size at which it triggers?